pkg/storage/stores/series_store_write.go

```diff
@@ -81,6 +87,13 @@ func (c *Writer) PutOne(ctx context.Context, from, through model.Time, chk chunk
 	if len(found) > 0 && !overlap {
 		writeChunk = false
 		DedupedChunksTotal.Inc()
+		encoded, err := chk.Encoded()
```

Comment:
one more thing: you'll be doing one extra encoding for every PutOne call. Isn't that expensive? WDYT of maybe having a feature flag/configuration to enable/disable this?